src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -83,6 +96,60 @@ private partial Assembly Load(AssemblyName assemblyName, string assemblyOriginal
             return loadContext.LoadFromAssemblyName(assemblyName);
         }
 
+        /// <summary>
+        /// Is this a registered analyzer file path that the loader knows about.
+        /// 
+        /// Note: this is using resolved paths, not the original file paths
+        /// </summary>
+        private bool IsRegisteredAnalyzerPath(string analyzerPath)
+        {
+            CheckIfDisposed();
+
+            lock (_guard)
+            {
+                return _resolvedToOriginalPathMap.ContainsKey(analyzerPath);
+            }
+        }
+
+        internal string? GetAssemblyLoadPath(AssemblyName assemblyName, string directory)
+        {
+            // Prefer registered dependencies in the same directory first.
+            var simpleName = assemblyName.Name!;
+            var assemblyPath = Path.Combine(directory, simpleName + ".dll");
+            if (IsRegisteredAnalyzerPath(assemblyPath))
+            {
+                return assemblyPath;
+            }
+
+            // Next if this is a resource assembly for a known assembly then load it from the 
+            // appropriate sub directory if it exists
+            //
+            // Note: when loading from disk the .NET runtime has a fallback step that will handle
+            // satellite assembly loading if the call to Load(satelliteAssemblyName) fails. This
+            // loader has a mode where it loads from Stream though and the runtime will not handle
+            // that automatically. Rather than bifurcate our loading behavior between Disk and
+            // Stream both modes just handle satellite loading directly
+            if (assemblyName.CultureInfo is not null && simpleName.EndsWith(".resources", SimpleNameComparer.Comparison))
```

Comment:
Looks like we have a similar logic for satellite assemblies duplicated between the Core and Desktop implementations. Could it be shared?

Reply:
There is a subtle difference between the two implementations that makes them hard to share. In the core version we _know_ the directory that contains the assembly name that we are probing for. In the desktop version we have to _guess_ by inferring a directory name from an assembly name.